FEATURE: updates discourse_theme to latest core changes

jjaffeux opened the pull request:
- adds missing template (package.json, eslint, ...)
- follows new js structure
- uses new apiInitializer
- minor refactor of cli prompts into UI class
- asks for component or not and builds template accordingly
- asks for author name
- removes a code duplication

davidtaylorhq commented on lib/discourse_theme/scaffold.rb:
```diff
@@ -1,75 +1,119 @@
+      };
+    STR
+
+    EN_YML = <<~STR
+      en:
+    STR
```
```suggestion
    EN_YML = <<~YAML
      en:
        #my_locale_key: My Translation
        theme_metadata:
          description: "My theme description"
          settings:
            # my_setting_name: My Setting Description
    YAML
```
(maybe we could prompt for a description?)

Using `YAML` rather than `STR` means that it gets syntax-highlighted nicely in most editors (including GitHub UI)

jjaffeux replied:
Cool stuff :+1:

davidtaylorhq commented on lib/discourse_theme/scaffold.rb:
```diff
@@ -1,75 +1,119 @@
+        UI.info "Creating .gitignore"
         File.write('.gitignore', GIT_IGNORE)
 
+        initializer = "javascripts/discourse/api-initializers/#{name}.js"
```
Should this be under the `/discourse` namespace? Or should it be under `/my-theme-name`? (either work... so not sure which we should be encouraging)

jjaffeux replied:
I personally use discourse, and like it better as it's following core pattern, but I don't mind

davidtaylorhq reviewed (approved):
Couple of comments, but overall looks good. The test failures will need checking.

jjaffeux committed:
FEATURE: updates discourse_theme to latest core changes

- adds missing template (package.json, eslint, ...)
- follows new js structure
- uses new apiInitializer
- minor refactor of cli prompts into UI class
- asks for component or not and builds template accordingly
- asks for author name
- removes a code duplication

jjaffeux committed:
also track gitignore

jjaffeux commented:
@davidtaylorhq I updated pr with requested changes

davidtaylorhq reviewed (approved):
Looks great to me!
